### PR TITLE
Add `tetra` problem

### PR DESCRIPTION
**Pull request by tmigot**

### CODE REVIEW

File: src/tetra.jl [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  x0 = xe_tetra,
+  TETS::Vector{Int64} = Tets_tetra,
+  Const::Vector{Int64} = Constants_tetra
+)
+  τ = 0.0
+  n = length(x0)
+  N = round(Int, n / 3)
+  E = round(Int, length(TETS) / 4)
+
+  lvar = -Inf * ones(n)
```
Comment (frapac) [inline src/tetra.jl]: I would maybe use directly `lvar = fill(-Inf, n)`

File: src/tetra.jl [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  lvar[Const] = x0[Const]
```
Comment (frapac) [inline src/tetra.jl]: Shouldn't we use the broadcast operator there?

File: src/tetra.jl [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  lvar[Const .+ N] = x0[Const .+ N]
+  lvar[Const .+ 2 * N] = x0[Const .+ 2 * N]
+
+  uvar = Inf * ones(n)
```
Comment (frapac) [inline src/tetra.jl]: similarly, I would maybe use `uvar = fill(Inf, n)`

File: src/tetra.jl [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            (x[TETS[e + 3 * E] + N * mod(i + 1, 3)] - x[TETS[e] + N * mod(i + 1, 3)])
+          ) *
+          sqrt(2) for i = 0:2
+        ))^(2 / 3)
+      ) for e = 1:E
+    )
+  )
+
+  for e = 1:E
+    @constraint(
```
Comment (frapac) [inline src/tetra.jl]: How about defining all the constraints in one block? 
```julia
@constraint(
    nlp, 
    [e in 1:E],
    exprs[e], 
)
```

File: src/tetra.jl [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+  return nlp
+end
+
+tetra_duct12_model() = tetra_model(xe_duct12, TETS_duct12, Const_duct12)
+tetra_duct15_model() = tetra_model(xe_duct15, TETS_duct15, Const_duct15)
+tetra_duct20_model() = tetra_model(xe_duct20, TETS_duct20, Const_duct20)
+tetra_hook_model() = tetra_model(xe_hook, TETS_hook, Const_hook)
+tetra_foam5() = tetra_model(xe_foam5, TETS_foam5, Const_foam5)
+tetra_gear() = tetra_model(xe_gear, TETS_gear, Const_gear)
```
Comment (frapac) [inline src/tetra.jl]: Shouldn't we rename the functions as `tetra_gear_model` and `tetra_foam5_model` for consistency?

**Review by frapac — COMMENTED** [review]
Thank you! it looks like we are almost done implementing the whole COPS benchmark :) 

I just have a few minor comments. Otherwise, the PR looks good to merge, as usual!

**Comment by tmigot** [comment]
Thanks for the feedback! I will open the last one `minsurfo` next :)

**Review by frapac — APPROVED** [review]

**Pull request merged**